source/code/source/playfab/PlayFabEventPipeline.cpp

```diff
@@ -267,7 +274,7 @@ namespace PlayFab
             }
             else
             {
-                auto requestBatchPtr = std::shared_ptr<const std::vector<std::shared_ptr<const IPlayFabEmitEventRequest>>>(new std::vector<std::shared_ptr<const IPlayFabEmitEventRequest>>(std::move(foundBatchIterator->second)));
+                auto requestBatchPtr = std::shared_ptr<const std::vector<std::shared_ptr<const IPlayFabEmitEventRequest>>>(new std::vector<std::shared_ptr<const IPlayFabEmitEventRequest>>(foundBatchIterator->second));
```

Comment:
>std::shared_ptr<const std::vector<std::shared_ptr<const IPlayFabEmitEventRequest>>> [](start = 39, length = 83)

this might just be a team code culture difference, but from my perspective this line is VERY hard to parse.

I think a good first step might be to use `std::make_shared` to simplify the `new` portion of this expression 

`std::make_shared<const std::vector<std::shared_ptr<const IPlayFabEmitEventRequest>>>(std::move(foundBatchIterator->second))`

in the longer term, it's pretty crazy to me that such a long type declaration is necessary and signals a pretty complicated interface. We had talked about it on friday, but this seems like a prime place to ask why all of the std::shared_ptr stuff is necessary. For now anything that can be done to make this stuff easier to read is goodness in my opinion!